draft-pauly-ohai-svcb-config.md

```diff
@@ -73,69 +78,76 @@ document.
 
-The "ohttp-path" SvcParamKey {{iana}} is used to convey the URI path of
-the oblivious target to which oblivious HTTP requests can sent. In both
-wire format and presentation format, this is a UTF-8 encoded string
-that contains the path segment of a URI. If this path parameter is not
-present, oblivious requests can be made to the root "/" path.
-
 ## Use in HTTPS service records
 
 For the "https" scheme, which uses the HTTPS RR type instead of SVCB,
-the presence of the "ohttp-configs" parameter means that the service
+the presence of the "oblivious" parameter means that the service
 being described is an Oblivious HTTP service that uses the default
 "message/bhttp" media type {{OHTTP}}
 {{!BINARY-HTTP=I-D.draft-ietf-httpbis-binary-message}}.
 
-When present in an HTTPS record, the "ohttp-configs" MUST be included
-in the mandatory parameter list, to ensure that implementations that
-do not understand the key do not interpret this service as a generic
-HTTP service.
+For example, an HTTPS service record for svc.example.com that supports
+an oblivious target could look like this:
 
-Clients MUST validate that they can parse the value of "ohttp-configs"
-as a valid key configuration before attempting to use the service.
+~~~
+svc.example.com. 7200  IN HTTPS 1 . alpn=h2,h2 oblivious
+~~~
+
+A similar record for a service that only support oblivious connectivity
+could look like this:
+
+~~~
+oblivious-svc.example.com. 7200  IN HTTPS 1 . (
+    mandatory=oblivious oblivious )
+~~~
 
 ## Use in DNS server SVCB records
 
 For the "dns" scheme, as defined in {{DNS-SVCB}}, the presence of
-the "ohttp-configs" parameter means that the DNS server being
+the "oblivious" parameter means that the DNS server being
 described is an Oblivious DNS over HTTP (DoH) service. The default
 media type expected for use in Oblivious HTTP to DNS resolvers
 is "application/dns-message" {{!DOH=RFC8484}}.
 
-The "ohttp-configs" parameter is only defined for use with DoH, so
-the "alpn" SvcParamKey MUST indicate support for a version of HTTP
-and the "dohpath" SvcParamKey MUST be present. The "ohttp-configs"
-MUST also be included in the mandatory parameter list, to ensure
-that implementations that do not understand the key do not interpret
-this service as a generic DoH service.
-
-Clients MUST validate that they can parse the value of "ohttp-configs"
-as a valid key configuration before attempting to use the service.
-
 ### Use with DDR {#ddr}
 
 Clients can discover an oblivious DNS server configuration using
 DDR, by either querying _dns.resolver.arpa to a locally configured
 resolver or querying using the name of a resolver {{DDR}}.
 
+For example, a DoH service advertised over DDR can be annotated
+as supporting oblivious resolution using the following record:
+
+~~~
+_dns.resolver.arpa  7200  IN SVCB 1 doh.example.net (
+     alpn=h2 dohpath=/dns-query{?dns} oblivious )
+~~~
+
 In the case of oblivious DNS servers, the client might not be able to
```

Comment:
The line does is no longer relevant. The client will anyway have to connect to the oblivious target to discover the key config.

Reply:
No, it is still relevant — it may not direct connect to the IP address, it can go through a proxy.

Reply:
If the client uses a proxy (e.g., HTTP CONNECT), it can still validate the TLS server certificate of the oblivious DNS server. I prefer the client validating the target certificate itself rather than relying on the oblivious proxy to do the validation on its behalf.

Reply:
Ah, I see. I've re-written this paragraph to just say to do the normal TLS cert check!